app/kuma-dp/pkg/dataplane/readiness/component.go

```diff
@@ -0,0 +1,120 @@
+package readiness
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"sync/atomic"
+	"time"
+
+	"github.com/bakito/go-log-logr-adapter/adapter"
+	"github.com/pkg/errors"
+
+	"github.com/kumahq/kuma/pkg/core"
+	"github.com/kumahq/kuma/pkg/core/runtime/component"
+)
+
+const (
+	pathPrefixReady  = "/ready"
+	stateReady       = "READY"
+	stateTerminating = "TERMINATING"
+)
+
+// Reporter reports the health status of this Kuma Dataplane Proxy
+type Reporter struct {
+	socketPath    string
+	isTerminating atomic.Bool
+}
+
+var logger = core.Log.WithName("readiness")
+
+func NewReporter(socketPath string) *Reporter {
+	return &Reporter{
+		socketPath: socketPath,
+	}
+}
+
+func (r *Reporter) Start(stop <-chan struct{}) error {
+	_, err := os.Stat(r.socketPath)
+	if err == nil {
+		// File is accessible try to rename it to verify it is not open
```

Comment:
we already do similar things in

https://github.com/kumahq/kuma/blob/5e7b4f86e2b454131e707c3cb35ba740b5715573/app/kuma-dp/pkg/dataplane/metrics/server.go#L142 

should we dry it up?